#!/bin/sh /usr/share/dpatch/dpatch-run
## 01-pointer-targets-differ-in-signedness.dpatch by Fabian Greffrath <fabian@debian-unofficial.org>
##
## DP: Fixes: "pointer targets in passing argument 3 of ‘video_putstring’ differ in signedness"

@DPATCH@
diff -urNad invaders-1.0.0~/game.c invaders-1.0.0/game.c
--- invaders-1.0.0~/game.c	2008-01-17 15:22:16.000000000 +0100
+++ invaders-1.0.0/game.c	2008-01-18 08:02:12.000000000 +0100
@@ -197,9 +197,9 @@
   bool pressed;
 
   video_usecolor(6,1);
-  video_putstring(30,10,"  GAME OVER  ");
-  video_putstring(30,11,winner ? "  YOU WIN  " : "  YOU LOOSE  ");
-  video_putstring(30,12,"  PRESS ESC  ");
+  video_putstring(30,10,(uint8 *) "  GAME OVER  ");
+  video_putstring(30,11,(uint8 *) (winner ? "   YOU WIN   " : "  YOU LOSE   "));
+  video_putstring(30,12,(uint8 *) "  PRESS ESC  ");
   video_update();
   do{
     key_decode(&key,&pressed);
